### PR TITLE
Refactor test_parquet.py to use check_round_trip at module level

### PR DESCRIPTION
**Pull request by maximveksler**
Refactoring and unification of testing approach in `test_parquet.py` module. 

Iteration upon work that was done for https://github.com/pandas-dev/pandas/pull/19135#issuecomment-358161165

- [x] passes `git diff upstream/master -u -- "*.py" | flake8 --diff`
- [x] tests added / passed

### CODE REVIEW

File: pandas/tests/io/test_parquet.py [hunk 1]
```diff
@@ -110,48 +110,72 @@ def df_full():
...
-        df_compat.to_parquet('foo', 'bar')
+    if write_kwargs is None:
+        write_kwargs = {'compression': None}
 
+    if read_kwargs is None:
+        read_kwargs = {}
 
-def test_options_py(df_compat, pa):
-    # use the set option
+    if engine:
```
Comment (jreback) [inline pandas/tests/io/test_parquet.py]: is there a case where engine is not required?

File: pandas/tests/io/test_parquet.py [hunk 2]
```diff
@@ -110,48 +110,72 @@ def df_full():
                                pd.Timestamp('20130103')]})
 
 
-def test_invalid_engine(df_compat):
+def check_round_trip(df, engine=None, path=None,
+                     write_kwargs=None, read_kwargs=None,
+                     expected=None, check_names=True):
```
Comment (jreback) [inline pandas/tests/io/test_parquet.py]: can you add a proper doc-string here

File: pandas/tests/io/test_parquet.py [hunk 3]
```diff
@@ -110,48 +110,72 @@ def df_full():
...
-    df = df_compat
-    with tm.ensure_clean() as path:
+    if expected is None:
+        expected = df
 
-        with pd.option_context('io.parquet.engine', 'pyarrow'):
-            df.to_parquet(path)
+    if path is None:
+        with tm.ensure_clean() as path:
+            df.to_parquet(path, **write_kwargs)
```
Comment (jreback) [inline pandas/tests/io/test_parquet.py]: can you make both of these into a loop with a kwarg in the signature ``repeat=2``

```
for _ in range(repeat):
    .....
```

**Review by jreback — CHANGES_REQUESTED** [review]

File: pandas/tests/io/test_parquet.py (hunk 1 above)
Comment (maximveksler) [reply pandas/tests/io/test_parquet.py]: Yes, mostly when testing for "defaults" handling (i.e. default engine, or engine being configured via `option_context`. Please see here https://github.com/pandas-dev/pandas/pull/19332/files#diff-539eb41a29c68cc27818db4b480c2fd6R163

File: pandas/tests/io/test_parquet.py (hunk 3 above)
Comment (maximveksler) [reply pandas/tests/io/test_parquet.py]: Thanks... 

Python is not native for me, I was looking for a nicer way to do that 👍

File: pandas/tests/io/test_parquet.py [hunk 4]
```diff
@@ -110,48 +110,82 @@ def df_full():
...
+                     expected=None, check_names=True,
+                     repeat=2):
+    """
+    Verify parquet serialize and deserialize produce the same results.
+
+    Performs a pandas to disk and disk to pandas round trip,
+    then compares the 2 resulting DataFrames to verify full
+    cycle is successful.
+
+    :param df: Dataframe to be serialized to disk
```
Comment (jreback) [inline pandas/tests/io/test_parquet.py]: pls use the numpy-doc format (just at any other doc-string)

**Review by jreback — CHANGES_REQUESTED** [review]

File: pandas/tests/io/test_parquet.py (hunk 4 above)
Comment (jreback) [reply pandas/tests/io/test_parquet.py]: include Parameters, Raises sections

**Review by jreback — CHANGES_REQUESTED** [review]

File: pandas/tests/io/test_parquet.py [hunk 5]
```diff
@@ -110,48 +110,77 @@ def df_full():
...
+        Expected deserialization result
+    check_names: list of str, optional
+        Closed set of column names to be compared
+    repeat: int, optional
+        How many times to repeat the test
+    """
+
+    write_kwargs = write_kwargs or {'compression': None}
+    read_kwargs = read_kwargs or {}
+    expected = expected or df
```
Comment (jreback) [inline pandas/tests/io/test_parquet.py]:
```
if expected is None:
    expected = df
```

File: pandas/tests/io/test_parquet.py [hunk 6]
```diff
@@ -110,48 +110,77 @@ def df_full():
...
+    Parameters
+    ----------
+    df: Dataframe
+    engine: str, optional
+        'pyarrow' or 'fastparquet'
+    path: str, optional
+    write_kwargs: dict of str:str, optional
+    read_kwargs: dict of str:str, optional
+    expected: DataFrame, optional
+        Expected deserialization result
```
Comment (jreback) [inline pandas/tests/io/test_parquet.py]: say otherwise will be equal to the input

**Review by jreback — CHANGES_REQUESTED** [review]

**Review by jreback — APPROVED** [review]

**Comment by jreback** [comment]
thanks @maximveksler

**Pull request merged**